lib/iex/lib/iex/autocomplete.ex

```diff
@@ -437,43 +437,50 @@ defmodule IEx.Autocomplete do
       not ensure_loaded?(mod) ->
         []
 
-      docs = Code.get_docs(mod, :callback_docs) ->
-        Enum.map(docs, &elem(&1, 0))
+      docs = get_docs(mod, [:callback, :macrocallback]) ->
+        Enum.map(docs, &extract_name_and_arity/1)
 
       true ->
         exports(mod)
     end
   end
 
+  defp get_docs(mod, kinds) do
+    case Code.fetch_docs(mod) do
+      {:docs_v1, _, _, _, _, _, docs} ->
+        for {{kind, _, _}, _, _, _, _} = doc <- docs, kind in kinds, do: doc
+
+      {:error, _} ->
+        nil
+    end
+  end
+
+  defp extract_name_and_arity({{_, name, arity}, _, _, _, _}), do: {name, arity}
+
   defp default_arg_functions_with_doc_false(docs) do
-    for {{fun_name, arity}, _, _, args, false} <- docs,
-        count = count_defaults(args),
+    for {{_, fun_name, arity}, _, signature, :hidden, _} <- docs,
+        count = count_defaults(signature),
         count > 0,
         new_arity <- (arity - count)..arity,
         do: {fun_name, new_arity}
   end
 
-  defp count_defaults(args) do
-    Enum.count(args, &match?({:\\, _, _}, &1))
+  defp count_defaults(signature) do
+    signature
+    |> Stream.flat_map(&Regex.scan(~r/ \\\\ /, &1))
```

Comment:
I'm not sure that's reliable - I could imagine code like this that would miscalculate here:
```elixir
iex(3)> quote(do: def foo(bar \\ ~S|\\|)) |> Macro.to_string
"def(foo(bar \\\\ ~S\"\\\\\"))"
```

Reply:
This is just temporary until we move the number of defaults to metadata. :)

Reply:
Yes, you're right. We discussed this earlier:

I wrote:

> I think we could replace [`count_defaults`] by adding something like `defaults: 2` or `defaults: [:year, :options]` to the metadata in `elixir_erl`

@josevalim:

> Regarding the defaults, the plan was to indeed at them to the metadata. So
I would leave the duplication for now and when we make it easier to define
the metadata from the Elixir side we can revisit it.

I would look at that after we remove `ExDc` and we have a chance to refactor the chunk generation and how and what we store about docs in ETS during compilation. I think we can live with this edge case in autocompletion until then.